modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/ObjectEntryOpenAPIContributor.java

```diff
@@ -278,6 +278,13 @@ private PathItem _createObjectRelationshipPathItem(
 		if (operation != null) {
 			return new PathItem() {
 				{
+					if (operations.get(HttpMethod.DELETE) != null) {
```

Comment:
I think it is a bit weird to add the Delete information just in case the Put method exists, because you are creating a dependency between them.

I would refactor to make it independant, for example, with the following code:

```java
private PathItem _createObjectRelationshipPathItem(
	PathItem existingPathItem, ObjectRelationship objectRelationship,
	String schemaName) {

	PathItem pathItem = new PathItem();

	Map<PathItem.HttpMethod, Operation> operations =
		existingPathItem.readOperationsMap();

	if (operations.containsKey(PathItem.HttpMethod.DELETE)) {
		pathItem.delete(
			_getObjectRelationshipDeleteOperation(
				objectRelationship, existingPathItem.getDelete(),
				schemaName));
	}

	if (operations.containsKey(PathItem.HttpMethod.GET)) {
		pathItem.get(
			_getObjectRelationshipGetOperation(
				objectRelationship, existingPathItem.getGet(), schemaName));
	}

	if (operations.containsKey(PathItem.HttpMethod.PUT)) {
		pathItem.put(
			_getObjectRelationshipPutOperation(
				objectRelationship, existingPathItem.getPut(), schemaName));
	}

	return pathItem;
}
```

What do you think about it?

Reply:
Makes perfect sense!! @ccorreagg